### PR TITLE
Add option to run tests in one folder

### PR DESCRIPTION
**Pull request by draupnirgudmundsson**
Added option to run tests in one schema folder.
Added option to assert subErrors (recursive). Not enabled by default because 70 tests fail when asserting subErrors but at least one can check their tests before submitting.
Added usage of the new options to the readme.md

### CODE REVIEW

**Comment by Tiano2017** [comment]
@draupnirgudmundsson thanks a lot for the contribution. I also made some changes to the test script recently. Would you mind sync with master and resolve the conflicts?

**Comment by draupnirgudmundsson** [comment]
@Tiano2017 I have now synced with master and changed how the parameters are set.
Please review.

File: tools/runSchemaValidation.js [hunk 1]
```diff
@@ -105,14 +105,14 @@ describe('run schema tests - ', () => {
...
-				}
-			}
+			// for (const testcase of testResult.testcases) {
+			// 	if (!testcase.valid) {
+			// 		message += `\n`;
+			// 		message += `\n     Test file: ${testcase.testFile}`;
+			// 		message += `\n     Test name: ${testcase.testName}`;
+			// 		message += `\n     ${testcase.message}`;
+			// 	}
+			// }
```
Comment (TianoMS) [inline tools/runSchemaValidation.js]: why are these  commented out?

Comment (TianoMS) [reply tools/runSchemaValidation.js]: this is actually for the CI build of the PR. if you look at "All checks" at the bottom of your PR, you can get CI run for your PR. and you can see the result if you click on it. 

You can also run the test by running "npm test" in  "/tools" folder locally. 

---
In reply to: [245772133](https://github.com/Azure/azure-resource-manager-schemas/pull/542#discussion_r245772133) [](ancestors = 245772133)

**Comment by TianoMS** [comment]
> 		let testResult = schemaTest02.runSchemaTests();

this one runs all the tests. I think we should still list out the result. 

---
Refers to: tools/runSchemaValidation.js:103 in 8194523. [](commit_id = 81945231ac3e184c43b8ad23e0acde9298d9a21a, deletion_comment = False)

File: tools/runSchemaValidation.js (hunk 1 above)
Comment (draupnirgudmundsson) [reply tools/runSchemaValidation.js]: Sorry this was from my debugging. Will update in a minute

**Comment by draupnirgudmundsson** [comment]
@TianoMS I have fixed what I accidentally committed. Sorry for that.

**Comment by Tiano2017** [comment]
PR looks good now!

**Pull request merged**